IPA-End-to-End.md

```diff
@@ -120,15 +120,28 @@ These events will occur on different _websites/apps_, so they then need to be ga
+There are two types of queries which can be issued by a _report collector_, each designed to provide information to a single site:
 
-Conversely, a _trigger fanout query_ is composed of _trigger reports_ from a single _trigger website/app_, and _source reports_ from multiple _source websites/apps_. When a _report collector_, acting on behalf of a specific _source_/_trigger websites/app_, issues a query, the query must either be a _source fanout query_ with _source reports_ from that _website/app_, or a _trigger fanout query_ with _trigger reports_ from that _website/app_.
+* A _source fanout query_ is designed to help a _website/app_ understand the effect that the ads it shows have on outcomes for its advertisers.  A _source fanout query_ can only contain _source reports_ from a single _source website/app_.  A _source fanout query_ can include _trigger reports_ from multiple sites, where each report might represent a conversion event.
 
-When making a _fanout query_, the _report collector_ uses the information it receives about events to decide whether to include an event in the query and what value is attached to the event.  The _report collector_ also assigns a _breakdown key_ to each _source event_ and a _trigger value_ to each _trigger value_.  The _report collector_ also assigns a value of its choice for the _attribution constraint ID_ of all events.
+* A _trigger fanout query_ helps a site that buys ads to understand how its advertising is helping to drive outcomes on its _website/app_.  A _trigger fanout query_ can only contain _trigger reports_ from a single _source website/app_.  A _trigger fanout query_ includes _source reports_ from multiple sites, where each report might represent an ad impression or click.
+
+The _report collector_ assigns a number of values to _source reports_ and _trigger reports_ when making a query.  The values that are assigned to each report determine what the query results mean:
+
+* A _breakdown key_ is associated with each _source report_ in a query.  The _breakdown key_ is used to split _source reports_ into groups for aggregation.  A _breakdown key_ can represent an advertising campaign, an advertising channel, a set of creatives, or any combination of these with other data from the event.  The output of the query includes an aggregate value for each _breakdown key_.
+
+* A _trigger value_ is associated with each _trigger report_.  If the _trigger report_ is matched with a _source reports_ by the MPC, the _trigger value_ contributes to the aggregated value for the _breakdown key_ from the _source report_.  Note that [sensitivity capping](#user-level-sensitivity-capping) might mean that some _trigger values_ do not contribute to the final result.
```

Comment:
```suggestion
* A _trigger value_ is associated with each _trigger report_.  If the _trigger report_ is matched with a _source reports_ by the MPC, the _trigger value_ contributes to the aggregated value for the _breakdown key_ from the _source report_.  Note that [sensitivity capping](#user-level-sensitivity-capping) might mean that some _trigger values_ do not contribute to the final result.

* A _timestamp_ is associated with each _source report_ and each _trigger report_. These timestamps are used to validate that we only attribute events when the _trigger report timestamp_ is after the _source report timestamp_. We are likely to also consider more sophisticated operations, such as checking that timestamps are within a fixed relative distance, e.g., a day or a week.

```

Reply:
Maybe just:

> * A _timestamp_ for each report.

I don't think that we need to go into details about validation and whatnot just yet.

Reply:
Sure, that's fine. I was mostly just following the form of the others 🙂